packages/hint-compat-api/src/compat-api-css-next.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @fileoverview Hint to validate if the CSS features of the project are not broadly supported
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, ProblemLocation } from 'hint/dist/src/lib/types';
+import { StyleParse, StyleEvents } from '@hint/parser-css/dist/src/types';
+import { CompatApi, userBrowsers, CompatCSS } from './helpers';
+import { BrowserSupportCollection } from './types';
+import { SimpleSupportStatement, SupportStatement } from './types-mdn.temp';
+import { browserVersions } from './helpers/normalize-version';
+
+import meta from './meta/compat-api-css';
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+type UserPrefixes = {
+    [key: string]: boolean;
+};
+
+export default class implements IHint {
+    public static readonly meta = meta;
+
+    private mdnBrowsersCollection: BrowserSupportCollection;
+    private compatApi: CompatApi;
+    private compatCSS: CompatCSS;
+    private userPrefixes: UserPrefixes = {};
+
+    public constructor(context: HintContext<StyleEvents>) {
+        const isCheckingNotBroadlySupported = true;
+
+        this.mdnBrowsersCollection = userBrowsers.convert(context.targetedBrowsers);
+        this.compatApi = new CompatApi('css', this.mdnBrowsersCollection, isCheckingNotBroadlySupported);
+        this.compatCSS = new CompatCSS(context, (...params) => {
+            this.testFeatureIsSupportedInBrowser(...params);
+        });
+
+        context.on('parse::end::css', async (styleParse: StyleParse) => {
+            await this.onParseCSS(styleParse);
+        });
+    }
+
+    private async onParseCSS(styleParse: StyleParse): Promise<void> {
+        const { resource } = styleParse;
+
+        this.compatCSS.setResource(resource);
+        await this.compatCSS.searchCSSFeatures(this.compatApi.compatDataApi, this.mdnBrowsersCollection, styleParse);
+    }
+
+    private async testFeatureIsSupportedInBrowser(browsersToSupport: BrowserSupportCollection, browserToSupportName: string, browserInfo: SupportStatement, featureName: string, prefix?: string, location?: ProblemLocation): Promise<void> {
```

Comment:
There are few methods with a lot of parameters (like this one).

What do you think about having a type "Feature" with all the data needed?

Reply:
We are thinking in something like
FeatureInfo (featureName: string, prefix?: string, location?: ProblemLocation)
BrowsersInfo (browsersToSupport: BrowserSupportCollection, browserToSupportName: string, browserInfo: SupportStatement)

Reply:
Implemented as part of #14

Reply:
as part of #14?

>We are thinking in something like
FeatureInfo (featureName: string, prefix?: string, location?: ProblemLocation)
BrowsersInfo (browsersToSupport: BrowserSupportCollection, browserToSupportName: string, browserInfo: SupportStatement)

Looks good to me

Reply:
> as part of #14?
My mistake, I meant https://github.com/CKGrafico/hint/pull/14